applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py

```diff
@@ -50,26 +69,32 @@ def AnalyzeDesignAndReportToCommunicator( self, currentDesign, optimizationItera
         step_before_analysis = optimization_model_part.ProcessInfo.GetValue(KM.STEP)
         delta_time_before_analysis = optimization_model_part.ProcessInfo.GetValue(KM.DELTA_TIME)
 
+        response_id = 1

```

Comment:
I see now that distinguishing between objective and constraint is not easily possible here, so the idea of the counter does not work that well. I would use the `identifier` (key of self.response_functions) instead.

Reply:
i'll change :)

Reply:
done :)